flake.nix

```diff
@@ -0,0 +1,59 @@
+{
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-23.11";
+
+    fenix = {
+      url = "github:nix-community/fenix";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+
+    flake-compat = {
+      url = "github:edolstra/flake-compat";
+      flake = false;
+    };
+  };
+
+  outputs = { self, nixpkgs, fenix, ... }:
+    let
+      inherit (nixpkgs) lib;
+
+      supportedSystems = [
+        "x86_64-linux"
+        "aarch64-linux"
+        "x86_64-darwin"
+        "aarch64-darwin"
+      ];
+
+      eachSupportedSystem = lib.genAttrs supportedSystems;
+
+      mkDevShells = system:
+        let
+          pkgs = import nixpkgs { inherit system; };
+
+          rust-toolchain = fenix.packages.${system}.fromToolchainFile {
+            file = ./rust-toolchain.toml;
+            sha256 = "opUgs6ckUQCyDxcB9Wy51pqhd0MPGHUVbwRKKPGiwZU=";
+          };
+
+          defaultShell = pkgs.mkShell {
+            buildInputs = with pkgs; [
+              # Nix
+              nixd
```

Comment:
It's one of the main stream Nix language server for Nix. It can definitely be removed if you want a minimalist version. It just convenient for Nix (+ direnv) users to enter the project and have a LSP ready to use, including the code formatter.